echo.go

```diff
@@ -530,7 +530,7 @@ func (e *Echo) Reverse(name string, params ...interface{}) string {
 
 // Routes returns the registered routes.
 func (e *Echo) Routes() []*Route {
-	routes := []*Route{}
```

Comment:
This statement should be fine as we are reassigning the variable.

Reply:
`routes := make([]*Route, 0, len(e.router.routes))`

this is alloc space for this part:

```go
 	for _, v := range e.router.routes {
    routes = append(routes, v)
  }
```

Reply:
If you preallocate it then would it be simpler to do this instead:

```go
routes := make([]*Route, len(e.router.routes))

for i, v := range e.router.routes {
    routes[i] = v
}
```

You can avoid the append call.

Reply:
@alexaandru `e.router.routes` is `map[string]*Route`
that's why `i` is a `string`, not an `integer`

Reply:
Looked into append function so it uses the preallocated capacity, so we are good 👍